lib/ts/Diags.cc

```diff
@@ -926,3 +917,29 @@ Diags::rebind_stderr(int new_fd)
   }
   return false;
 }
+
+void
+Diags::vprintline(FILE *fp, char *buffer, va_list ap) const
+{
+  int nbytes = strlen(buffer);
+
+  // check if we need to scrub any patterns
+  if (unlikely(scrub_enabled)) {
+    char _buf[MAX_LOG_LINE_SIZE];
+    int r = vsnprintf(_buf, MAX_LOG_LINE_SIZE, buffer, ap);
+    if (r > MAX_LOG_LINE_SIZE) {
+      sprintf(_buf, "[WARN] Diags log line too long, dropping log\n");
+    } else if (r < 0) {
+      sprintf(_buf, "[WARN] Diags log line failed to encode with vsnprintf\n");
+    }
+    scrubber->scrub_buffer(_buf);
+    fprintf(fp, "%s", _buf);
```

Comment:
Should probably just `strlcpy` here, although the `fprintf` is likely almost as fast.